app/controllers/users_controller.rb

```diff
@@ -0,0 +1,40 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new(session[:user_params])
+    session.delete(:user_params)
+  end
+
+  def create
+    user = User.new(user_params)
+    if user.valid? && passwords_match?
+      user.save
+      flash[:success] = "You are now registered and logged in!"
+      redirect_to "/profile"
+    else
+      flash[:failure] = "All fields must be completed before submitting:"
```

Comment:
Is creating customized flash types this easy? For earlier versions of Rails, you had to insert add_flash_type into ApplicationController. That's cool - good to know.